allow null for framework.translator.default_path

SimonHeimberg opened the pull request:
Allow to configure framework.translator.default_path to null, as it was until symfony 3.4.x (fix BC compatibility).

| Q             | A
| ------------- | ---
| Branch?       | 4.4 for bug fixes <!-- see below -->
| Bug fix?      | yes
| New feature?  | no <!-- please update src/**/CHANGELOG.md files -->
| Deprecations? | no <!-- please update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tickets       | Fix #37111 <!-- prefix each issue number with "Fix #", no need to create an issue if none exist, explain below instead -->
| License       | MIT
| Doc PR        | no <!-- required for new features -->

stloyd commented on src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:
```diff
@@ -1213,7 +1213,9 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
         }
 
-        if ($container->fileExists($defaultDir)) {
+        if (null == $defaultDir) {
```
```suggestion
        if (null === $defaultDir) {
```

SimonHeimberg replied:
Oops, sorry. Thank you. It looks like I write in too many programming languages. I will commit your suggestion as soon as the way to go is decided.

SimonHeimberg replied:
I had to fix this once again, once I knew it by heart, but I miss some training ...

ro0NL commented:
shouldnt we fix it at the Configuration level, given `->defaultValue('%kernel.project_dir%/translations')`

IIUC `cannotBeEmpty` is missing.

SimonHeimberg commented:
> shouldnt we fix it at the Configuration level, given `->defaultValue('%kernel.project_dir%/translations')`
> 
> IIUC `cannotBeEmpty` is missing.

I see two options:
* allow null again, as it was before 3.4.x (not sure about the exact version)
* prevent null cleanly (Your suggestion is probably cleaner. Failing with a fatal php error as currently is bad.)

I am not sure which way to go. Actually it was a BC break in 3.4.x, which was not noticed quickly. For more details see  #37111

ro0NL commented:
i think it broke as of 4.x due the added type hint in ContainerBuilder::fileExists

before the inexistence was implicit, eg. `if ($container->fileExists($dir = $defaultDir.'/'.$name)) {` checks `/$name` at root level if $defaultDir is null.

So i think allowing explicit `null` in config is reasonable and should be handled as such, everywhere. As it preserves BC with 3.4

SimonHeimberg committed:
allow null for framework.translator.default_path

Allow to configure framework.translator.default_path to null, as it was until symfony 3.4.x (fix BC compatibility).

fixes #37111

SimonHeimberg commented:
Should I add a test case checking this config value? Where should I add it? I doubt that `src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/ConfigurationTest.php` would test enough places :thinking:

fabpot commented:
Thank you @SimonHeimberg.
